nisystemlink/clients/dataframe/_data_frame_client.py

```diff
@@ -119,6 +120,31 @@ def query_tables(self, query: models.QueryTablesRequest) -> models.PagedTables:
         """
         ...
 
+    def query_tables_generator(
+        self, query: models.QueryTablesRequest
+    ) -> Generator[models.TableMetadata, None, None]:
+        """Queries available tables on the SystemLink DataFrame service and returns their metadata.
+
+        Args:
+            query: The request to query tables. `continuation_token` is ignored.
+
+        Yields:
+            `models.TableMetadata` Table Metadata
+        """
+        _query = query.copy()
+        _query.continuation_token = None
+        while True:
+            paged_tables = self.query_tables(query=_query)
```

Comment:
Should we include exponential backoff on this so that we protect users from running into rate limiting when they use this style for large queries?